program/src/state.rs

```diff
@@ -1102,7 +1103,32 @@ impl HealthCache {
         };
 
         let h = self.health[health_type as usize].ok_or(throw!())?;
-        Ok(h + curr_perp_health - prev_perp_health)
+
+        // Apply taker fees; Assume no referrer
+        let taker_fees = if taker_quote != 0 {
+            let taker_quote_native =
+                I80F48::from_num(info.quote_lot_size.checked_mul(taker_quote.abs()).unwrap());
+            match mango_group.find_token_index(&mngo_token::id()) {
+                None => info.taker_fee * taker_quote_native,
+                Some(mngo_index) => {
+                    let mngo_cache = &mango_cache.root_bank_cache[mngo_index];
```

Comment:
do we need to check cache liveness here?

Reply:
Not necessary because if user had any MNGO it would be in his ActiveAssets and it would have been checked already